helix-view/src/document.rs

```diff
@@ -910,4 +1064,244 @@ mod test {
             ]
         );
     }
+
+    #[test]
+    fn test_big5_decode() {
```

Comment:
Can you make a macro that takes a label and expands to a whole test fn, then use it to define all these encoding tests? It's equivalent to a table test